### PR TITLE
Use new CircleCI convenience image for PHP

### PR DESCRIPTION
**Pull request by byrond**
See https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034

Todo:

- [x] Update `defaults/install/.circleci/config.yml`
- [x] Maybe add a comment with a link to the orb documentation for installing browsers?
- [x] ~Maybe add a comment about when the mysql image will get a cimg version?~ Update for new MySQL image.
- [x] The devel version issue should be resolved in the `develop` branch now

### CODE REVIEW

**Comment by becw** [comment]
I added some todos to the description but this change sounds good!

**Comment by byrond** [comment]
@becw, even though https://github.com/palantirnet/the-build/pull/176 is merged, Composer grabs the latest release of The Build (which is 4.0-alpha4) during install. That version doesn't have the Devel fix, so https://github.com/palantirnet/drupal-skeleton/pull/119 will fail until there is a new release of 4.x of The Build.

[Edited to correct the installed version in the skeleton and clarify that the release doesn't need to be stable.]

**Comment by byrond** [comment]
@becw, I created https://github.com/palantirnet/the-build/releases/tag/4.0-alpha5 and https://github.com/palantirnet/drupal-skeleton/pull/119 is passing now without specifying the dev branch.

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,14 +24,10 @@ jobs:
           command: |
             sudo apt-get update
             sudo apt-get install -y libpng-dev default-mysql-client
```
Comment (byrond) [inline .circleci/config.yml]: This probably needs a `sudo apt-get update && ` to match https://github.com/palantirnet/drupal-skeleton/pull/119/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R25

Comment (byrond) [reply .circleci/config.yml]: Well, it already has one right above, but I moved them to the same line to stay consistent with other config.yml files.

**Comment by byrond** [comment]
@becw I just made one last update to the example deployment config snippets. Otherwise, I think this is ready to go.

**Review by becw — APPROVED** [review]

**Pull request merged**